_user_guides/armory-pipeline-templates.md

```diff
@@ -0,0 +1,187 @@
+
+ They are JSON files with replacable values in them. e.g., a module that defines a wait stage in a pipeline might look like:
+```
+{
+    "name": "Wait",
+    "refId": "1",
+    "requisiteStageRefIds": [],
+    "type": "wait",
+    "waitTime": 42
+}
+```
+- **Pipeline definitions**: These define a pipeline for an application in a file called `dinghyfile`. The `dinghyfile` usually resides at the root level of the application repo. eg:
+
+![dinghyfile](http://f.cl.ly/items/3t3z0Q2Z040f0i0V2P3O/dinghyfile.png)
+
+You can compose stage/task templates to make a full definition. e.g., a Pipeline definition for a spinnaker application called `foo` that has a single wait stage might look like:
+```
+{
+  "application": "foo",
+  "pipelines": [
+    {
+      "application": "foo",
+      "keepWaitingPipelines": false,
+      "limitConcurrent": true,
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {
+          "name": "Wait",
+          "refId": "1",
+          "requisiteStageRefIds": [],
+          "type": "wait",
+          "waitTime": 10
+        }
+      ],
+      "triggers": []
+    }
+  ]
+}
+```
+
+## Template variables and substitution
+
+We can have Pipeline definitions use Modules defined in another GitHub Repo. e.g.:
+```{% raw %}
+{
+  "application": "foo",
+  "pipelines": [
+    {
+      "application": "foo",
+      "keepWaitingPipelines": false,
+      "limitConcurrent": true,
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {{ module "wait.stage.module" }} // Module created in dinghy-templates repo
+      ],
+      "triggers": []
+    }
+  ]
+}
+{% endraw %}```
+We can also overwrite variables inside the imported module like so:
+```{% raw %}
+{
+  "application": "foo",
+  "pipelines": [
+    {
+      "application": "foo",
+      "keepWaitingPipelines": false,
+      "limitConcurrent": true,
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {{ module "wait.stage.module" "waitTime" 200 }}
+      ],
+      "triggers": []
+    }
+  ]
+}
+{% endraw %}```
+Any number of variables can be overwritten in the same module by simply specifying them as arguments. e.g.: `{% raw %}{{ module "wait.stage.module" "waitTime" 100 "name" "simpleWait" }}{% endraw %}`.
+
+> Note: We do not support complex data-type variable substitution in the alpha release
+
+Let us create a more realistic pipeline using templates. One that would look like this:
+
+![demopipeline](http://f.cl.ly/items/1z3z3Z2w3j2w35171U39/Screen%20Shot%202018-03-12%20at%2011.18.38%20AM.png)
+
+You would use the following JSON to create such. Note that any of the stages could have come from an imported module, but we show the full JSON here for readability:
+
+```
+{
+  "application": "demo",
+  "pipelines": [
+    {
+      "application": "demo",
+      "keepWaitingPipelines": false,
+      "limitConcurrent": true,
+      "name": "step1",
+      "stages": [
```

Comment:
the only other improvement I might suggest is swap/add one of these stages to to modules.

@LeSandeep said there's a bug with having templates specifying `requisiteStageRefIds` because dinghy doesn't support arrays at the moment.